AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin.swift

```diff
@@ -57,6 +57,9 @@ final public class AWSAPIPlugin: NSObject, APICategoryPlugin {
         }
     }
 
+    /// Lock used for performing operations atomically when getting and setting `reachabilityMap`.
+    let reachabilityMapLock = NSLock()
```

Comment:
nit: prefer to assign properties in the same place; since we have an `init` for this type, let's assign the value there instead of making it hard to find initial value assignments
```suggestion
    let reachabilityMapLock: NSLock
```